src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -62,7 +61,7 @@ public TermsQueryParser() {
 
     @Override
     public String[] names() {
-        return new String[]{NAME, "in"};
+        return new String[]{TermsQueryBuilder.NAME, TermsLookupQueryBuilder.NAME, "in"};
```

Comment:
I'd rather merge TermsQueryBuilder and TermsLookupBuilder, this is going to be a problem anyway when registering builders for serialization if we want to keep registering the parser only and deduce the builder from it . I don't see any value in having two builders for the same query, it becomes confusing for java api users too.

Also, the easy way of creating queries should be through querybuilders, we can keep the existing `QueryBuilders#termsLookupQuery` but it will return a TermsQueryBuilder instead. We can also add more methods to `QueryBuilders` to create a terms lookup query that holds all the needed params.